app/src/main/java/com/example/firstapplication/ToDoListViewModel.kt

```diff
@@ -0,0 +1,25 @@
+package com.example.firstapplication
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.async
+
+class ToDoListViewModel(
+    private val repository: ToDoListRepository
+): ViewModel() {
+
+    fun getToDoList(userId: String): LiveData<List<ToDo>> {
+        return MutableLiveData<List<ToDo>>().also { liveData ->
```

Comment:
Chyba nie do końca załapałaś ideę LiveData. To jest analogiczny byt do Observable z Rx albo Flow z kotlina - czyli strumień danych, który można obserwować. Ty tworzysz za każdym razem nową instancję (Mutable)LiveData, pakujesz do niej dane raz i później nie aktualizujesz. A chodzi o to, by z viewModel wystawić jedną instancję, w której będziesz aktualizować wartość gdy zmienią się dane (np. dociągniesz z backandu albo coś się zmieni pod wpływem interakcji użytkownika)